sensors/simsensor_common/event_queue.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Queue for buffering sensor event data from reader to simsensors
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "event_queue.h"
+
+#include <stdlib.h>
+
+
+struct event_queue_t {
+	sensor_event_t *queue;
+	unsigned int queue_sz;
+	unsigned int act_elem_cnt;
+
+	sensor_event_t *head;
+	sensor_event_t *tail;
+};
+
+
+event_queue_t *eventQueue_init(unsigned int elem_cnt)
+{
+	event_queue_t *result;
+
+	if (elem_cnt == 0) {
+		return NULL;
+	}
+
+	result = malloc(sizeof(event_queue_t));
+	if (result == NULL) {
+		return NULL;
+	}
+
+	result->queue = malloc(sizeof(sensor_event_t) * elem_cnt);
+	if (result->queue == NULL) {
+		free(result);
+		return NULL;
+	}
+
+	result->queue_sz = elem_cnt;
+	result->act_elem_cnt = 0;
+
+	result->head = result->queue;
+	result->tail = result->queue;
+
+	return result;
+}
+
+
+void eventQueue_free(event_queue_t *q)
+{
+	if (q == NULL) {
+		return;
+	}
+
+	free(q->queue);
+	free(q);
+}
+
+
+static sensor_event_t *eventQueue_nextElement(event_queue_t *q, sensor_event_t *elementPtr)
+{
+	if (elementPtr == q->queue + q->queue_sz) {
+		return q->queue;
+	}
+
+	return elementPtr + 1;
+}
```

Comment:
This is more of a code remark.

Let's try to keep things simple. More verbose naming should be used when the goal of the named object is not easily understandable from the context. For example: here we are in the queue module and we have a function that accepts some pointer and returns some pointer to the container type with the word `next` in the name. It is easily understandable that it may _return the next pointer to the passed pointer_. How about that:
```suggestion
static inline sensor_event_t *eventQueue_next(event_queue_t *q, sensor_event_t *elem)
{
	return (elem == q->queue + q->queue_sz) ? q->queue : elem + 1;
}
```
Shorter names benefit with overall shorter line lengths which helps readibility and allow for less lines in total without loosing clarity :)